 Analysis of Variance (ANOVA)
 
Research Question:

    Is there relation of crater depth and depend on crater diameter image?

Null Hypothesis (H0):

    Crater depth is not associated with crater diameter image.

   


Alternative Hypothesis (H1):

    Crater depth is some times associated with crater diameter image.

 

In [1]:
from __future__ import print_function
from statsmodels.compat import urlopen
import numpy as np
np.set_printoptions(precision=4, suppress=True)
import statsmodels.api as sm
import pandas as pd
pd.set_option("display.width", 100)
import matplotlib.pyplot as plt
from statsmodels.formula.api import ols
import statsmodels.formula.api as smf
from statsmodels.graphics.api import interaction_plot, abline_plot
from statsmodels.stats.anova import anova_lm
import statsmodels.stats.multicomp as multi 
data = pd.read_csv("crateres.csv")


M = "DIAM_CIRCLE_IMAGE"
S = "NUMBER_LAYERS"

sub1 = data[(data["DIAM_CIRCLE_IMAGE"]>=50) & (data["DIAM_CIRCLE_IMAGE"]<=85)]
sub2 = sub1.copy()

model1 = smf.ols(formula='DIAM_CIRCLE_IMAGE ~ C(DEPTH_RIMFLOOR_TOPOG)', data=data)
results1 = model1.fit()
print (results1.summary())


sub2 = sub1[['DIAM_CIRCLE_IMAGE', 'DEPTH_RIMFLOOR_TOPOG']].dropna()


print ('means for DIAM_CIRCLE_IMAGE by major DEPTH_RIMFLOOR')
m1= sub2.groupby('DEPTH_RIMFLOOR_TOPOG').mean()
print (m1)

print ('standard deviations for DIAM_CIRCLE_IMAGE by major DEPTH_RIMFLOOR')
m2= sub2.groupby('DEPTH_RIMFLOOR_TOPOG').std()
print (m2)

print ('means for NUMBER_LAYERS by major dDEPTH_RIMFLOOR sub1 data')
m3= sub1.groupby('NUMBER_LAYERS').mean()
print (m3)

print ('means for NUMBER_LAYERS by major dDEPTH_RIMFLOORsub1 data')
m4= sub1.groupby('MORPHOLOGY_EJECTA_1').mean()
print (m4)               




#mc1 = multi.MultiComparison(sub1['NUMBER_LAYERS'], sub1['DIAM_CIRCLE_IMAGE'])
#res1 = mc1.tukeyhsd()
#print(res1.summary())


                            OLS Regression Results                            
Dep. Variable:      DIAM_CIRCLE_IMAGE   R-squared:                       0.489
Model:                            OLS   Adj. R-squared:                  0.479
Method:                 Least Squares   F-statistic:                     47.36
Date:                Fri, 20 May 2016   Prob (F-statistic):               0.00
Time:                        20:35:30   Log-Likelihood:                -13768.
No. Observations:                4999   AIC:                         2.774e+04
Df Residuals:                    4899   BIC:                         2.839e+04
Df Model:                          99                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [95.0% Conf. Int.]
----------------------------------------------------------------------------------------------

The ANOVA model indicated that the F value is high (47.36)